sdk/metric/controller/basic/controller.go

```diff
@@ -216,23 +249,29 @@ func (c *Controller) collect(ctx context.Context) error {
 // compute the CheckpointSet.  This applies the configured collection
 // timeout.  Note that this does not try to cancel a Collect or Export
 // when Stop() is called.
-func (c *Controller) checkpoint(ctx context.Context, cond func() bool) error {
-	ckpt := c.checkpointer.CheckpointSet()
+func (c *Controller) checkpoint(ctx context.Context) error {
+	for _, impl := range c.provider.List() {
+		if err := c.checkpoint1(ctx, impl.(*accumulatorCheckpointer)); err != nil {
```

Comment:
Do we want this type assertion to panic?

Reply:
The panic would only result from a bug. The provider is a registry owned by this object that will call this code to produce the corresponding value, so it didn't seem necessary to test the error condition.

I've added an "impossible type assertion failed: %T" to to defend against the impossible (in two places) 😁